src/app/dim-ui/CharacterSelect.test.tsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+import CharacterSelect from './CharacterSelect';
+import { DimStore } from '../inventory/store-types';
+import { Frame, Track, ViewPager } from 'react-view-pager';
+import SimpleCharacterTile from 'app/inventory/SimpleCharacterTile';
+
+let mockDimStore = jest.genMockFromModule('../inventory/store-types') as DimStore;
+let mockDimStoreArray: DimStore[] = [];
+let dimStore1 = jest.genMockFromModule('../inventory/store-types') as DimStore;
+let dimStore2 = jest.genMockFromModule('../inventory/store-types') as DimStore;
+let dimStore3 = jest.genMockFromModule('../inventory/store-types') as DimStore;
+const mockOnCharacterChanged = jest.fn();
+
+beforeEach(() => {
```

Comment:
I'm a little bit wary of mocks, in general. Could these `DimStores` have just been made as objects instead?

Reply:
They could have, but then we're writing more integration tests than anything. I started out by storing some `DimStore` objects inside files in the solution itself, but each one had about 10 MB of data inside of it and reading it from disk was slow.

Why are you wary of mocks? I chose this solution because the `DimStore` objects are very large and reading them from disk for each test suite would end up being incredibly slow; I believe it took 20 seconds to read a fully formed `DimStore` object into memory for one test suite. Getting them from Bungie's API isn't ideal, either.

I don't know how to generate `DimStore` objects without calling Bungie's API or having very large files stored on disk. If you know of an easier way to generate them, I could try that and let you know.

Reply:
They're pretty much just plain objects - you can just build one up. You don't have to a full character with a thousand items.

If this ends up being difficult I'd just skip testing these bits.

Reply:
Created empty objects instead of mocks for `DimStores`